NeoML/src/TraditionalML/BytePairEncoder.cpp

```diff
@@ -175,6 +164,37 @@ void CBytePairEncoder::Serialize( CArchive& archive )
 	}
 }
 
+void CBytePairEncoder::LoadDictionary( const CWordDictionary& _tokens, bool _useEndOfWordToken, bool _useStartOfWordToken )
+{
+	useEndOfWordToken = _useEndOfWordToken;
+	useStartOfWordToken = _useStartOfWordToken;
+
+	tokenToId.DeleteAll();
+	tokens.DeleteAll();
+	ClearCache();
+
+	// Sort words descending
+	CWordDictionary finalizedDictionary;
+	_tokens.CopyTo( finalizedDictionary );
+
+	finalizedDictionary.Finalize( INT64_MIN );
+	for( int i = 0; i < finalizedDictionary.Size(); i++ ) {
+		const CString newToken = finalizedDictionary.GetWord( i );
+		NeoAssert( !tokenToId.Has( newToken ) );
+		tokenToId.Add( newToken, tokens.Size() );
+		tokens.Add( newToken );
+	}
+}
+
+void CBytePairEncoder::GetDictionary( CWordDictionary& output ) const
+{
+	output.Empty();
+	output.AddWord( UnkToken, 0 );

```

Comment:
Maybe 0 can be replaced by tokens.Size() + 1? And then count towards 0.

So we can save property of non-negative frequencies without `assert` statements deletion (See below).